### ISSUE TITLE
portal visitors see owner's posts first

### ISSUE THREAD

**Issue by eelfroth** [Rotonde/rotonde-client#68]
currently, when you visit another user's Rotonde page, you see their feed much like they would see it. This makes it difficult to see what they themselves have posted. my proposal is, that by default the feed is filtered by the portal's @name if you are not the owner of the page.
to make it possible to still see their full feed (including posts by others that don't mention them) if one so desires, I made it so that filters get removed immediately upon clicking the "filtering by …" message. this way, you can remove filters even when you don't have an operator.

→PR #63 
following @ianh 's advice to discuss the matter in a github issue.
you can test the proposed changes yourself by going to my portal: dat://5276ca2f92b6d508933ac4b01ef9078938f69bed9ca15c462617f531641b7a6b/

**Comment by frozenpandaman** [comment]
this is a good change imo. while it's cool to be able to see how *others'* homepages/feeds look like, if i navigate to their page, i typically want to see the posts by them (at least by default – can always toggle it off easily!)

**Comment by ianh** [comment]
I agree this is a useful feature, but I also find it nice to explore other people's pages as they see them (without having to click through a filter link at the top).  I wonder if there's a more subtle way to focus self-entries on the page, like:

- showing a "pinned entry" that you can choose
- showing your last few entries at the top or on the side
- adding a way to scroll between self-entries
- or something else…

**Comment by frozenpandaman** [comment]
Semi-related, once a timeline "fills up" with more recent stuff (specifically, 42 entries + the $rotonde "bot" post) there's no way to see older stuff. Should we allow for multiple pages or even pagination/infinite scrolling so you can see all the way back? I feel like if I'm gone for a day, as long as I'm following more than a couple people, I'll miss out on mentions/whispers.

**Comment by ianh** [comment]
Yeah—you can always filter to see mentions, but I think just nobody has gotten around to implementing pagination yet.  We should probably have another issue for it.

**Comment by frozenpandaman** [comment]
Oh, that's true—I forget about that. Thanks for the reminder of that feature. :) I'll make an issue for it, at least to provide a platform for related discussion.

**Comment by eelfroth** [comment]
@ianh my intention was to implement the feature in a minimally-invasive way, utilizing the already in place filtering system. the problem with filtering right now is just that you can't do it on other people's pages. toggle-able filtering by portalname gives visitors some rudimentary filter capabilities.
(sidenote: filtering unowned pages works fine from the dev console)

your suggestions are cool features in their own right. pinned posts would be very nice! would we have to change the specs to allow for a 'pinned' field in portal.json?

**Comment by ianh** [comment]
It wouldn't necessarily have to be in portal.json, since other clients don't care about it.  We could have a "local.json" to store things like whether a post is pinned, for example.

**Comment by forresto** [comment]
Showing "local" posts first will speed up the first view load, right?

**Comment by aeonofdiscord** [comment]
Not the way it's currently implemented; the client is still loading all the data from everybody's portals, it's just filtering them after the fact so only one person's posts are visible. Lazy-loading is an interesting idea, though.

**Comment by neauoire** [comment]
Sort of related, I've implemented something pretty neat that works like this: 
```
dat://2f21e3c122ef0f2555d3a99497710cd875c7b0383f998a2d37c02c042d598485/#neauoire-98
```
Will link directly to a specific entry, it could be modified so `#neauoire` displays all posts by user.
